src/controllers/wallet/wallet.controller.js

```diff
@@ -259,15 +259,37 @@ async function decodeTx(req, res) {
     return;
   }
 
-  const { txHex } = req.body;
+  const txHex = req.body.txHex || null;
+  const partialTx = req.body.partial_tx || null;
+
+  if (txHex && partialTx) {
+    res.status(400).json({
+      success: false,
+      error: 'Required only one of txHex or partialTx',
+    });
+    return;
+  }
+
   try {
-    const tx = helpersUtils.createTxFromHex(txHex, req.wallet.getNetworkObject());
+    let tx;
+    if (txHex !== null) {
+      tx = helpersUtils.createTxFromHex(txHex, req.wallet.getNetworkObject());
+    } else {
+      const partial = PartialTx.deserialize(partialTx, req.wallet.getNetworkObject());
+      // Validate will check with the fullnode if the partial-tx inputs exists and are valid
+      const valid = await partial.validate();
+      if (!valid) {
+        throw new Error('Partial transaction inconsistent with backend');
```

Comment:
I don't think we should throw an error here, maybe just 

```
res.status(400).json({
      success: false,
      error: 'Invalid partial transaction hex',
    });
    return;
```

What do you think?

Reply:
No, I think it's good